application/controllers/ApiV1ContactsController.php

```diff
@@ -0,0 +1,604 @@
+        }
+
+        $method = $request->getMethod();
+        if (
+            in_array($method, ['POST', 'PUT'])
+            && (
+                ! preg_match('/([^;]*);?/', $request->getHeader('Content-Type'), $matches)
+                || $matches[1] !== 'application/json'
+            )
+        ) {
+            $this->httpBadRequest('No JSON content');
+        }
+
+        $results = [];
+        $responseCode = 200;
+        $db = Database::get();
+
+        /** @var ?string $identifier */
+        $identifier = $request->getParam('identifier');
+
+        if ($identifier && ! Uuid::isValid($identifier)) {
+            $this->httpBadRequest('The given identifier is not a valid UUID');
+        }
+
+        $filter = FilterProcessor::assembleFilter(
+            QueryString::fromString(rawurldecode(Url::fromRequest()->getQueryString()))
+                ->on(
+                    QueryString::ON_CONDITION,
+                    function (Filter\Condition $condition) {
+                        $column = $condition->getColumn();
+                        if (! in_array($column, ['id', 'full_name', 'username'])) {
+                            $this->httpBadRequest(sprintf(
+                                'Invalid filter column %s given, only id, full_name and username are allowed',
+                                $column
+                            ));
+                        }
+
+                        if ($column === 'id') {
+                            if (! Uuid::isValid($condition->getValue())) {
+                                $this->httpBadRequest('The given filter id is not a valid UUID');
+                            }
+
+                            $condition->setColumn('external_uuid');
+                        }
+                    }
+                )->parse()
+        );
+
+        switch ($method) {
+            case 'GET':
+                $stmt = (new Select())
+                    ->distinct()
+                    ->from('contact co')
+                    ->columns([
+                        'contact_id'        => 'co.id',
+                        'id'                => 'co.external_uuid',
+                        'full_name',
+                        'username',
+                        'default_channel'   => 'ch.name',
+                    ])
+                    ->joinLeft('contact_address ca', 'ca.contact_id = co.id')
+                    ->joinLeft('channel ch', 'ch.id = co.default_channel_id');
+
+                if ($identifier !== null) {
+                    $stmt->where(['external_uuid = ?' => $identifier]);
+
+                    /** @var stdClass|false $result */
+                    $result = $db->fetchOne($stmt);
+
+                    if ($result === false) {
+                        $this->httpNotFound('Contact not found');
+                    }
+
+                    if ($result->username === null) {
+                        unset($result->username);
+                    }
```

Comment:
Same as for the groups. Do not omit the property but return it with its empty state.